src/Analyzers/Core/Analyzers/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs

```diff
@@ -90,44 +109,73 @@ protected sealed override void InitializeWorker(AnalysisContext context)
+            var fieldReads = s_fieldToUsageLocationPool.Allocate();
+            var fieldWrites = s_fieldToUsageLocationPool.Allocate();
+
+            // Record the names of all the private fields in this type.  We can use this to greatly reduce the amount of
             // binding we need to perform when looking for restrictions in the type.
+            var fieldNames = _fieldNamesPool.Allocate();
             foreach (var member in namedType.GetMembers())
             {
-                if (member is IFieldSymbol field)
+                if (member is IFieldSymbol
+                    {
+                        // Can only convert fields that are private, as otherwise we don't know how they may be used
+                        // outside of this type.
+                        DeclaredAccessibility: Accessibility.Private,
+                        // Only care about actual user-defined fields, not compiler generated ones.
+                        CanBeReferencedByName: true,
+                        // Will never convert a constant into an auto-prop
+                        IsConst: false,
+                        // Can't preserve volatile semantics on a property.
+                        IsVolatile: false,
+                        // To make processing later on easier, limit to well-behaved fields (versus having multiple
+                        // fields merged together in error recoery scenarios).
+                        DeclaringSyntaxReferences.Length: 1,
+                    } field)
+                {
                     fieldNames.Add(field.Name);
+                }
             }
 
-            context.RegisterSyntaxNodeAction(context => AnalyzePropertyDeclaration(context, namedType, analysisResults), PropertyDeclarationKind);
+            // Examine each property-declaration we find within this named type to see if it looks like it can be converted.
+            context.RegisterSyntaxNodeAction(
+                context => AnalyzePropertyDeclaration(context, namedType, fieldNames, analysisResults),
+                PropertyDeclarationKind);
+
+            // Concurrently, examine the usages of the fields of this type within itself to see how those may impact if
+            // a field/prop pair can actually be converted.
             context.RegisterCodeBlockStartAction<TSyntaxKind>(context =>
             {
-                RegisterIneligibleFieldsAction(fieldNames, ineligibleFields, context.SemanticModel, context.CodeBlock, context.CancellationToken);
-                RegisterNonConstructorFieldWrites(fieldNames, nonConstructorFieldWrites, context.SemanticModel, context.CodeBlock, context.CancellationToken);
+                RecordIneligibleFieldLocations(fieldNames, ineligibleFields, context.SemanticModel, context.CodeBlock, context.CancellationToken);
+                RecordAllFieldReferences(fieldNames, fieldReads, fieldWrites, context.SemanticModel, context.CodeBlock, context.CancellationToken);
```

Comment:
note: it's possible we might be ablet to merge htis into one pass.  but i want to save that for a followup pr.